main.go

```diff
@@ -520,7 +611,30 @@ func mainAction(c *cli.Context) error {
 
 // beforeAction is triggered before other actions are triggered
 func beforeAction(c *cli.Context) error {
-	level, err := log.ParseLevel(c.String("log-level"))
+	/* Load the configuration values from the config file specified*/
+	filePath := c.String("config")
+	if filePath != "" {
+		inputSource, err := altsrc.NewTomlSourceFromFile(filePath)
+		if err != nil {
+			return err
+		}
+		if err := altsrc.ApplyInputSourceValues(c, inputSource, c.App.Flags); err != nil {
+			return err
+		}
+	}
+
+	DefaultConfig = Conf{
+		LogLevel: c.String(cliLogLevel),
+		CertFile: c.String(cliCertFile),
+		KeyFile:  c.String(cliKeyFile),
+	}
+
+	err := DefaultConfig.configureAPIServer()
+	if err != nil {
+		return err
```

Comment:
The default value should be provided by rhsm if this is not set. I will change the approach, instead of storing this information in the Conf struct, when it tries to select the API server hostname, first it will try to read this configuration. 

That way it would not cause any conflicts while building rhc or if the config file doesn't exists.